src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,165 @@
 package core.basesyntax;
 
+import java.util.Iterator;
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public void add(T value) {
+        if (first == null) {
```

Comment:
It could be more elegant
```suggestion
        Node<T> node = new Node<>(value, null, null);
        if (first == null) {
            first = last = node;
        } else {
            last.next = node;
            node.prev = last;
            last = node;
        }
        size++;
```